### DISCUSSION TITLE
Gamified leaderboard of Mattermost Interactions

### DISCUSSION THREAD

**Discussion by chetanyakan** [mattermost/community#75] [Practical Ideas]
A plugin that adds gamified elements to encourage users to use Mattermost more.

- Show this week's/month's/year's most active user on Mattermost in the form of a leaderboard.
    - You could use a DB query to calculate a count of interactions (posts, replies, at-mentions, reactions) and show a leaderboard of these interactions or plot graphs for whether user's engagement is increasing/decreasing over time.
- Use the Badges plugin to send a badge to the most active user for week/month etc.
- Show daily interaction streaks like at least one message a day for n days etc.

**Comment by Willyfrog** [comment]
This should be a fun idea.

Throwing some fun stats would be cool, like 

- Most used custom emoji.

- User most mentioned.

On a discord server I'm in, there is a bot that awards you xp (random number between 3-10 I believe) per message. So you can check your rank without actually knowing how many messages.
